### PR TITLE
users: cockpit.spawn's superuser option is not a boolean

### PR DESCRIPTION
**Pull request by jelly**
According to the docs and protocl the superuser option is "try" and "require". Split off from https://github.com/cockpit-project/cockpit/pull/20308 let's see what the tests say :)

### CODE REVIEW

File: pkg/users/users.js [hunk 1]
```diff
@@ -82,7 +82,7 @@ function AccountsPage() {
         const handleShadow = cockpit.file("/etc/shadow", { superuser: "try" });
         handleShadow.watch(() => debouncedGetLogins(), { read: false });
 
-        const handleLogindef = cockpit.file("/etc/login.defs", { superuser: true });
+        const handleLogindef = cockpit.file("/etc/login.defs", { superuser: "try" });
```
Comment (martinpitt) [inline pkg/users/users.js:85]: This is a behaviour change, but /etc/login.defs is world-readable and only used for watching -- why would it need superuser in the first place? But anyway, for the purpose of this PR it's fine.

**Review by martinpitt — APPROVED** [review]
Cheers! Some flakes, retrying, but LGTM.

File: pkg/users/users.js (hunk 1 above)
Comment (jelly) [reply pkg/users/users.js:85]: I rather we clean it up then and leave superuser out.

Comment (jelly) [reply pkg/users/users.js]: That's done now :)

**Review by martinpitt — APPROVED** [review]
Could have been a  separate commit, but ok -- if this goes green, I'm happy! Thanks!

**Review by allisonkarlitskaya — COMMENTED** [review]
There's the typo in the commit message (mentioned when reviewing it in the other PR).

I'm still not a big fan of how one commit mixes a straight-refactor with functionality changes (and doesn't mention the functionality change in the commit message).

**Review by allisonkarlitskaya — APPROVED** [review]
Looks good, thanks!

**Pull request merged**